src/transformStateWithClones.js

```diff
@@ -7,7 +8,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newObject = { ...state };
+  const result = [];
+  const ACTION_CASES = {
+    add: 'addProperties',
+    remove: 'removeProperties',
+    clear: 'clear',
+  };
+
+  actions.forEach(element => {
+    switch (element.type) {
```

Comment:
in order to make it more readable you can use destructuring :

`actions.forEach({ type, extraData, keysToRemove } => { `
 `const { add, remove, clear } = ACTION_CASES;`